src/typeDefs/mutations.ts

```diff
@@ -88,14 +88,7 @@ export const mutations = gql`
       uninstalledOrgs: [ID!]
     ): Plugin!
 
-    createAdvertisement(
-      orgId: ID!
-      name: String!
-      link: String!
-      type: String!
-      startDate: Date!
-      endDate: Date!
-    ): Advertisement!
+    createAdvertisement(data: CreateAdvertisementInput!): Advertisement!
```

Comment:
rename `data` to `input`, the point is to have uniform structure for mutations

here `<mutation_name>` is a placeholder
```graphql
input <mutation_name>Input {
  # any arbitrary object structure required by the mutation, can include scalars, enums and input types 
}

type <mutation_name>Payload {
  # any arbitrary object structure you want to return as the result of this mutation
  # usually would include a single field that the mutation works on, but can be extended to more than
  # one fields
}

type Mutation {
  <mutation_name>(input: <mutation_name(in PascalCase)>Input!): <mutation_name(in PascalCase)>Payload
}
```

here's a real world example, `<mutation_name>` is replaced with `createUser`, `<mutation_name(in PascalCase)>` is replaced with `CreateUser`
```graphql
type User {
  _id: ID!
  age: Int!
  name: String!
}

input CreateUserInput {
  age: Int!
  name: String!
}

type CreateUserPayload {
  accessToken: String!
  refreshToken: String!
  user: User!
}

type Mutation {
  createUser(input: CreateUserInput!): CreateUserPayload
}
```

you see how this would lead to a coherent graphql schema structure for mutations? we can always be sure that each mutation would:-

1. have at least zero and at most one input object defined for it
2. have either zero arguments or exactly one `nullable` argument named `input` with name starting as `mutation(in PascalCase)` and ending with `Input` appended to it
3. return exactly one object type with name starting as `mutation_name(in PascalCase)` and ending with `Payload` appended to it

if you go lax on these standards and conventions, you create a spaghetti hard to understand graphql schema

Reply:
Thanks for the detailed explanation.